replicateexistingcluster/replicateexistingcluster.go

```diff
@@ -0,0 +1,224 @@
+	}
+	pcs, err := cnvtPriorityClassesListToApplyConfigurationList(expRes.PriorityClasses)
+	if err != nil {
+		return nil, xerrors.Errorf("call cnvtPriorityClassesListToApplyConfigurationList: %w", err)
+	}
+	return &export.ResourcesForImport{
+		Pods:            pods,
+		Nodes:           nodes,
+		Pvs:             pvs,
+		Pvcs:            pvcs,
+		StorageClasses:  scs,
+		PriorityClasses: pcs,
+		// existingClusterExportService can't export the SchedulerConfig.
+		SchedulerConfig: nil,
+	}, nil
+}
+
+func cnvtPodListToApplyConfigurationList(pods []corev1.Pod) ([]v1.PodApplyConfiguration, error) {
+	rto := make([]v1.PodApplyConfiguration, len(pods))
+	for i, p := range pods {
+		if err := convertToApplyConfiguration(p, &rto[i]); err != nil {
+			return nil, xerrors.Errorf("convert Pod to apply configuration: %w", err)
+		}
+	}
+	return rto, nil
+}
+
+func cnvtNodeListToApplyConfigurationList(nodes []corev1.Node) ([]v1.NodeApplyConfiguration, error) {
+	rto := make([]v1.NodeApplyConfiguration, len(nodes))
+	for i, n := range nodes {
+		if err := convertToApplyConfiguration(n, &rto[i]); err != nil {
+			return nil, xerrors.Errorf("convert Node to apply configuration: %w", err)
+		}
+	}
+	return rto, nil
+}
+
+func cnvtPvListToApplyConfigurationList(pvs []corev1.PersistentVolume) ([]v1.PersistentVolumeApplyConfiguration, error) {
+	rto := make([]v1.PersistentVolumeApplyConfiguration, len(pvs))
+	for i, p := range pvs {
+		if err := convertToApplyConfiguration(p, &rto[i]); err != nil {
+			return nil, xerrors.Errorf("convert PersistentVolume to apply configuration: %w", err)
+		}
+	}
+	return rto, nil
+}
+
+func cnvtPvcListToApplyConfigurationList(pvcs []corev1.PersistentVolumeClaim) ([]v1.PersistentVolumeClaimApplyConfiguration, error) {
+	rto := make([]v1.PersistentVolumeClaimApplyConfiguration, len(pvcs))
+	for i, p := range pvcs {
+		if err := convertToApplyConfiguration(p, &rto[i]); err != nil {
+			return nil, xerrors.Errorf("convert PersistentVolumeClaim to apply configuration: %w", err)
+		}
+	}
+	return rto, nil
+}
+
+func cnvtStorageClassesListToApplyConfigurationList(scs []storagev1.StorageClass) ([]cfgstoragev1.StorageClassApplyConfiguration, error) {
+	rto := make([]cfgstoragev1.StorageClassApplyConfiguration, len(scs))
+	for i, s := range scs {
+		if err := convertToApplyConfiguration(s, &rto[i]); err != nil {
+			return nil, xerrors.Errorf("convert StorageClass to apply configuration: %w", err)
+		}
+	}
+	return rto, nil
+}
+
+func cnvtPriorityClassesListToApplyConfigurationList(pcs []schedulingv1.PriorityClass) ([]schedulingcfgv1.PriorityClassApplyConfiguration, error) {
+	rto := make([]schedulingcfgv1.PriorityClassApplyConfiguration, len(pcs))
+	for i, p := range pcs {
+		if err := convertToApplyConfiguration(p, &rto[i]); err != nil {
+			return nil, xerrors.Errorf("convert PriorityClasses to apply configuration: %w", err)
+		}
+	}
+	return rto, nil
+}
+
+func convertToApplyConfiguration(in interface{}, out interface{}) error {
```

Comment:
```suggestion
// convertToApplyConfiguration is convert some object to XXXXApplyConfiguration.
// out should be the pointer of XXXXApplyConfiguration, otherwise you cannot get result of conversion.
//nolint:funlen // For readability.
func convertToApplyConfiguration(in interface{}, out interface{}) error {
```